### ISSUE TITLE
[TreeView] disabled node index problem 

### ISSUE THREAD

**Issue by Levon770** [telerik/kendo-angular#2335]
Getting following error, after selecting next sibling node of the disabled node.

```
Uncaught TypeError: Cannot read property 'disabled' of null
    at NavigationService.push../node_modules/@progress/kendo-angular-treeview/dist/fesm5/index.js.NavigationService.isDisabled (index.js:585)
    at TreeViewComponent.push../node_modules/@progress/kendo-angular-treeview/dist/fesm5/index.js.TreeViewComponent.clickHandler (index.js:1174)
    at HTMLElement.<anonymous> (platform-browser.js:993)
    at HTMLElement.globalListener (platform-browser.js:1268)
```

data

**To Reproduce**
The Html structure is

```
<kendo-treeview
                    [nodes]="treeData"
                    textField="name"
                    [hasChildren]="hasChildren"
                    [children]="fetchChildren"

                    [isExpanded]="isExpanded"
                    (collapse)="handleExpand($event)"
                    (expand)="handleExpand($event)"
                    (selectionChange)="selectTreeSegment($event)"
                    (nodeDblClick)="handleDblClick($event)"
                    kendoTreeViewSelectable
                    [(selectedKeys)]="selectedKeys"
                    [isDisabled]="isDisabled"
            >
                <ng-template kendoTreeViewNodeTemplate let-dataItem>
                    {{dataItem.name}}
                    <span class="badge badge-success" *ngIf="dataItem.isNew">new</span>
                    <span class="badge  badge-info" *ngIf="!dataItem.isNew && dataItem.isUpdated">edited</span>
                    <span class="badge  badge-danger" *ngIf="dataItem.isDeleted">deleted</span>
                </ng-template>
            </kendo-treeview>
```
And a methods are: 

```
public readonly hasChildren = (node: any): boolean => node.has_children;

    public readonly isExpanded = (dataItem: {[k: string]: any}): boolean => dataItem.isExpanded;
    
    public readonly isDisabled = (dataItem: {[k: string]: any}): boolean => dataItem.isDeleted;

    public readonly handleExpand = (node) => node.dataItem.isExpanded = !node.dataItem.isExpanded;

```

I dont know it will be helpfull or not, but this function return null, and tryinig to get `disabled` property of null.
```
NavigationModel.prototype.find = function (index, nodes) {
        var childLevel = this.childLevel(nodes);
        var indexToMatch = this.ib.indexForLevel(index, childLevel);
        var isLeaf = childLevel === this.ib.level(index);
        var node = nodes.find(function (n) { return n && n.index === indexToMatch; });
        if (!node) {
            return null;
        }
        return isLeaf ? node : this.find(index, node.children);
    };
``` 

in this function `index` argument is '0_4', nodes is array and there is no item with this index
 only '0_0', '0_1', '0_2', '0_3', '0_5', '0_6', '0_7'

**Browser**
 - OS: Windows 10
 - Browser: Google Chrome Version 74.0.3729.169

**Comment by ipeshev** [comment]
@Levon770 could you try to reproduce the issue on this [stackblitz sample](https://stackblitz.com/edit/angular-pkyxsb?file=app/app.component.ts) ? We are suspecting old version of @progress/kendo-angulat-treeview package , could you try to update to latest version ?

**Comment by Levon770** [comment]
> @Levon770 could you try to reproduce the issue on this [stackblitz sample](https://stackblitz.com/edit/angular-pkyxsb?file=app/app.component.ts) ? We are suspecting old version of @progress/kendo-angulat-treeview package , could you try to update to latest version ?

current version is @progress/kendo-angular-treeview v 3.1.2

**Comment by Levon770** [comment]
Here is StackBlitz
https://angular-e3shmn.stackblitz.io

In current example: select ACC-222, click on disable, trye to select ACC-333

**Comment by tsvetomir** [comment]
Fixed in [development builds](https://www.telerik.com/kendo-angular-ui/components/installation/development-builds/).

**Comment by tapopov** [comment]
Available in `kendo-angular-treeview@4.0.1`
